### DISCUSSION TITLE
what is the strategy of triton for running models in parallel, multi-thread or multi-process?

### DISCUSSION THREAD

**Discussion by heivens** [triton-inference-server/server#6253] [Q&A] [answered]
what is the strategy of triton for running models in parallel, multi-thread or multi-process?

**Comment by the-david-oy** [comment, marked as answer]
This differs based on the backend and model configuration. For example, Python backend runs models in their own processes. TensorRT uses CUDA streams. It also depends on your model configuration (e.g. if you specify multiple model instances, they will be on the same device, so many backends use multi-threading to enable parallel inference).

**Reply by varungv** [reply]
So for a python backend model having 10 instances, we essentially would be opening up 1 Process and 10 threads ? Is that correct ?
